### ISSUE TITLE
Casting Spells Without Mana

### ISSUE THREAD

**Issue by SoulMuncher** [jdoleary/Spellmasons#401]
Tested in headless, may not work in singleplayer

Example: If you don't have enough mana to cast two arrows, you can cast one arrow at a time in quick succession and it won't update your mana fast enough to cancel the 2nd spell message, so both arrows will go through. With more max mana to setup a buffer, some planning, and some practice, you can cast TONS of free spells every turn.

Related: #182 
This alone wouldn't fix the issue, but would help, and have other positive game effects.

Related: #399 
Might be good to prevent the player from sending multiple spell messages at once?

**Comment by SoulMuncher** [comment]
More context and repro examples:
https://discord.com/channels/1032294536640200766/1220843269999693956

**Comment by jdoleary** [comment]
More context: 
```
X1PrototypeOP
 — 
05/27/2024 4:12 AM
Hey guys! New to the discord and have been loving the game! Wanted to give feedback on one of my most recent runs, where i figured out how to spawn WAY too many ads without much penalty. If you are playing with other people, and they are in the midst of casting a spell, you can preemptively summon a minion as many times as you can fit in the casting time of the other player, with no negatives besides performance and a large cooldown on said minion before another summon. I didn't see any other post like this so just wanted you to be aware! This was very fun either way and I'm looking forward to more runs!
```

**Comment by SoulMuncher** [comment]
Mostly fixed by: #982
